src/main/java/core/screens/LevelGameplayController.java

```diff
@@ -0,0 +1,94 @@
+package core.screens;
+
+import com.badlogic.gdx.Gdx;
+import com.badlogic.gdx.Screen;
+import com.badlogic.gdx.graphics.Color;
+import com.badlogic.gdx.graphics.GL20;
+import com.badlogic.gdx.graphics.g2d.TextureAtlas;
+import com.badlogic.gdx.graphics.glutils.ShapeRenderer;
+import com.badlogic.gdx.math.Vector2;
+import com.badlogic.gdx.physics.box2d.BodyDef;
+import core.camera.CameraManager;
+import core.levels.LevelManager;
+import core.world.WorldEntity;
+import core.world.WorldManager;
+
+/**
+ * Runs the gameplay of a Level.
+ */
+public class LevelGameplayController implements Screen {
+
+    private final CameraManager cameraManager;
+    private ShapeRenderer shapeRenderer;
+    private final LevelManager levelManager;
+    private final WorldEntity player;
+
+    public LevelGameplayController(LevelManager levelManager) {
```

Comment:
Does this imply that every LevelGameplayController has its own levelManager? Might be the best way to do this but thought to point out that TA was encouraging only 1 manager.

Reply:
You can set the `level` property of the LevelManager, so we only need one

still, probably makes more sense to tell the LevelGameplayController which LevelState to use, and have it set up the LevelManager itself